core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsService.scala

```diff
@@ -547,6 +555,18 @@ class SubmissionsService(
       ps.inputs.filter(_.inputResolutions.forall(_.error.isEmpty))
     )
 
+  def getSetToDelete(
+    submissionRequest: SubmissionRequest
+  ): Option[AttributeEntityReference] =
+    submissionRequest.deleteEntityType match {
+      case Some(value) if value == submissionRequest.entityType.get =>
+        submissionRequest.deleteEntityName match {
+          case Some(name) if name == submissionRequest.entityName.get => Some(AttributeEntityReference(value, name))
+          case _                                                      => None
+        }
+      case _ => None
+    }
```

Comment:
instead of nesting, you can treat the type+name as a tuple:
```scala
    (submissionRequest.deleteEntityType, submissionRequest.deleteEntityName) match {
      case (Some(entityType), Some(entityName)) => ???
      case _ => ???
    }
```
and, instead of using `.get` (which can throw an error if called on a `None`), use `.contains`:
```scala
value == submissionRequest.entityType.get
```
becomes
```scala
submissionRequest.entityType.contains(value)
```